teps/0046-triggertemplate-parameter-escaping.md

```diff
@@ -0,0 +1,295 @@
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: TriggerTemplate
+metadata:
+  name: escaped-param
+spec:
+  params:
+  - name: title
+    description: The title from the incoming body
+    escape: true
+```
+
+And in the code, we'd apply the current "old escape" logic, i.e. replacing `"`
+with `\"`, only on the parameters marked as requiring to be escaped, but there's
+definitely scope for alternative escaping.
+
+## Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy.  Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+-->
+
+## Design Evaluation
+
+
+## Drawbacks
+
+I'm not aware of any, it's just altering existing behaviour to make it more
+flexible.
+
+It should allow dropping of the TriggerTemplate-wide annotation to indicate that
+all parameters should be escaped.
+
+## Alternatives
+
+When the fix for [#777](https://github.com/tektoncd/triggers/pull/823) was
+landed, the annotation-based approach was applied, this allowed existing
+templates to continue to work, but fixed the issue with over-escaping values.
+
+The annotation approach was considered temporary at that point, but we could
+consider this permanent.
+
+Another would be to change the escaping to be some sort of strategy based
+approach, e.g. `quotes`, `base64`, `json`, where these would link to functions
+that would accept a `[]byte` and return an escaped `[]byte`.
+
+One other option, which might warrant further consideration, is instead of
+changing the _param_ to be an escaped parameter, to change the way the
+parameters are templated into the template resources when rendering.
+
+For example, currently we support `$(param.name)` and replace that with the
+parameter name (possibly escaped), we could support `%(param.name)` which would
+indicate that the particular insertion into the template should be escaped.
+
+## Infrastructure Needed (optional)
+
+## Upgrade & Migration Strategy (optional)
+
+The default for escaping would be `false` (Go default for bool), so nothing
+would change unless explicitly applied.
+
+We could hopefully deprecate the old-escape annotation within a couple of
+releases, there's arguably even support for detecting the annotation and
+changing all parameters in the TriggerTemplate to require escaping (which is
+what the annotation currently indicates).
+
+## References (optional)
+
+<!--
+Use this section to add links to GitHub issues, other TEPs, design docs in Tekton
+shared drive, examples, etc. This is useful to refer back to any other related links
+to get more details.
+-->
```

Comment:
is it possible to include some examples of how other systems handle this kind of issue?